Source/Csla/Core/FieldManager/PropertyInfoManager.cs

```diff
@@ -70,47 +71,18 @@ internal static PropertyInfoList GetPropertyListCache(Type objectType)
       var cache = PropertyInfoCache;
 
 #if NET5_0_OR_GREATER
-      var found = cache.TryGetValue(objectType, out var listInfo);
-
-      var list = listInfo?.Item2;
-
-      if (!found)
+      var list = cache.GetOrAdd(objectType, type => 
```

Comment:
As far as I understand the doc the following will happen:
At lesat two hits the line. It gets checked if it's there. If not _all_ start creating the value.
When the value is finished it's again checked if it's there - this time within a lock. If not it's added. All following "adder" are checking and now getting a result. That means they discard their creation and use the already created one from that point on.

Reply:
To get the previous behavior then, it sounds like there needs to be a lock in the adder to ensure it only runs once.

Reply:
That's the question: Is it acceptable to have the add-functionality run multiple times even though all consumers are getting the same result.

Reply:
That's what I'm wondering. The pattern used throughout CSLA has proven to be quite stable and robust, and avoids the lock except in the very few possible hits during startup of a process.

If ConcurrentDictionary were to simplify that process, that's excellent. If it doesn't, then it isn't solving a problem - or is just requiring the same solution in a different form.

Reply:
Without taking a deep look into the `ConcurrentDictionary` implemention I can _imagine_ the whole `add` operation is less blocking than the current implemention. We are locking on `cache` so even unrelated opertions are paused until the `add` is finished eventhough the `key` does not have anything to do with it.
Tbh I don't see a great benefit on switching to the concurrent one. _Maybe_ it's a bit faster but as you said: This happens due to the static nature only once for every type.

Reply:
Maybe another view on that:
Pro `ConcurrentDictionary`
- less code
- Maybe faster due to better/optimized locking

Contra:
- Add func can be invoked multiple times

I personally like less code because it means less oportunities to introduce bugs. I somewhat can't imagine the difference is noticable. But I don't have hard data on that 🤷‍♂️

Reply:
The less code is always good. I think we need to add locking in the add function to avoid double-calls on initialization.

Reply:
Fixed